<html><head><meta content="text/html; charset=UTF-8" http-equiv="content-type"><style type="text/css">ol.lst-kix_q1ktesomj3tk-1.start{counter-reset:lst-ctn-kix_q1ktesomj3tk-1 0}.lst-kix_q1ktesomj3tk-0>li{counter-increment:lst-ctn-kix_q1ktesomj3tk-0}.lst-kix_q1ktesomj3tk-0>li:before{content:"" counter(lst-ctn-kix_q1ktesomj3tk-0,decimal) ". "}.lst-kix_q1ktesomj3tk-3>li{counter-increment:lst-ctn-kix_q1ktesomj3tk-3}ol.lst-kix_q1ktesomj3tk-8.start{counter-reset:lst-ctn-kix_q1ktesomj3tk-8 0}.lst-kix_q1ktesomj3tk-2>li:before{content:"" counter(lst-ctn-kix_q1ktesomj3tk-2,lower-roman) ". "}ol.lst-kix_q1ktesomj3tk-6.start{counter-reset:lst-ctn-kix_q1ktesomj3tk-6 0}.lst-kix_q1ktesomj3tk-1>li:before{content:"" counter(lst-ctn-kix_q1ktesomj3tk-1,lower-latin) ". "}.lst-kix_q1ktesomj3tk-7>li:before{content:"" counter(lst-ctn-kix_q1ktesomj3tk-7,lower-latin) ". "}.lst-kix_q1ktesomj3tk-6>li:before{content:"" counter(lst-ctn-kix_q1ktesomj3tk-6,decimal) ". "}.lst-kix_q1ktesomj3tk-8>li:before{content:"" counter(lst-ctn-kix_q1ktesomj3tk-8,lower-roman) ". "}ol.lst-kix_q1ktesomj3tk-3.start{counter-reset:lst-ctn-kix_q1ktesomj3tk-3 0}.lst-kix_q1ktesomj3tk-3>li:before{content:"" counter(lst-ctn-kix_q1ktesomj3tk-3,decimal) ". "}.lst-kix_q1ktesomj3tk-4>li:before{content:"" counter(lst-ctn-kix_q1ktesomj3tk-4,lower-latin) ". "}ol.lst-kix_q1ktesomj3tk-5.start{counter-reset:lst-ctn-kix_q1ktesomj3tk-5 0}.lst-kix_q1ktesomj3tk-5>li:before{content:"" counter(lst-ctn-kix_q1ktesomj3tk-5,lower-roman) ". "}.lst-kix_q1ktesomj3tk-6>li{counter-increment:lst-ctn-kix_q1ktesomj3tk-6}ol.lst-kix_q1ktesomj3tk-0.start{counter-reset:lst-ctn-kix_q1ktesomj3tk-0 0}ol.lst-kix_q1ktesomj3tk-7.start{counter-reset:lst-ctn-kix_q1ktesomj3tk-7 0}.lst-kix_q1ktesomj3tk-7>li{counter-increment:lst-ctn-kix_q1ktesomj3tk-7}.lst-kix_q1ktesomj3tk-2>li{counter-increment:lst-ctn-kix_q1ktesomj3tk-2}ol.lst-kix_q1ktesomj3tk-1{list-style-type:none}ol.lst-kix_q1ktesomj3tk-2.start{counter-reset:lst-ctn-kix_q1ktesomj3tk-2 0}ol.lst-kix_q1ktesomj3tk-0{list-style-type:none}ol.lst-kix_q1ktesomj3tk-3{list-style-type:none}ol.lst-kix_q1ktesomj3tk-4.start{counter-reset:lst-ctn-kix_q1ktesomj3tk-4 0}.lst-kix_q1ktesomj3tk-8>li{counter-increment:lst-ctn-kix_q1ktesomj3tk-8}ol.lst-kix_q1ktesomj3tk-2{list-style-type:none}ol.lst-kix_q1ktesomj3tk-5{list-style-type:none}.lst-kix_q1ktesomj3tk-5>li{counter-increment:lst-ctn-kix_q1ktesomj3tk-5}ol.lst-kix_q1ktesomj3tk-4{list-style-type:none}ol.lst-kix_q1ktesomj3tk-7{list-style-type:none}ol.lst-kix_q1ktesomj3tk-6{list-style-type:none}ol.lst-kix_q1ktesomj3tk-8{list-style-type:none}.lst-kix_q1ktesomj3tk-1>li{counter-increment:lst-ctn-kix_q1ktesomj3tk-1}.lst-kix_q1ktesomj3tk-4>li{counter-increment:lst-ctn-kix_q1ktesomj3tk-4}ol{margin:0;padding:0}table td,table th{padding:0}.c21{border-right-style:solid;padding:5pt 5pt 5pt 5pt;border-bottom-color:#000000;border-top-width:1pt;border-right-width:1pt;border-left-color:#000000;vertical-align:top;border-right-color:#000000;border-left-width:1pt;border-top-style:solid;border-left-style:solid;border-bottom-width:1pt;width:540pt;border-top-color:#000000;border-bottom-style:solid}.c29{border-right-style:solid;border-bottom-color:#000000;border-top-width:1pt;border-right-width:1pt;border-left-color:#000000;vertical-align:top;border-right-color:#000000;border-left-width:1pt;border-top-style:solid;border-left-style:solid;border-bottom-width:1pt;width:540pt;border-top-color:#000000;border-bottom-style:solid}.c3{padding-top:0pt;text-indent:36pt;padding-bottom:0pt;line-height:1.15;orphans:2;widows:2;text-align:left}.c10{-webkit-text-decoration-skip:none;color:#000000;font-weight:400;text-decoration:underline;text-decoration-skip-ink:none;font-size:11pt;font-style:normal}.c0{color:#000000;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:10pt;font-family:"Courier New";font-style:normal}.c15{margin-left:22pt;padding-top:0pt;text-indent:-22pt;padding-bottom:11pt;line-height:1.0;text-align:left}.c19{padding-top:0pt;padding-bottom:0pt;line-height:1.0;orphans:2;widows:2;text-align:left}.c4{padding-top:0pt;padding-bottom:0pt;line-height:1.15;orphans:2;widows:2;text-align:left}.c31{margin-left:22pt;padding-top:11pt;text-indent:-22pt;padding-bottom:11pt;line-height:1.0;text-align:left}.c2{font-size:11pt;font-style:italic;color:#000000;font-weight:400;text-decoration:none}.c5{font-size:11pt;font-style:normal;color:#000000;font-weight:400;text-decoration:none}.c16{color:#000000;text-decoration:none;font-size:11pt;font-style:normal}.c25{padding-top:0pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c32{text-decoration-skip-ink:none;-webkit-text-decoration-skip:none;color:#1155cc;text-decoration:underline}.c24{padding-top:0pt;padding-bottom:0pt;line-height:1.15;text-align:left}.c14{font-size:11pt;font-style:normal;color:#000000;font-weight:400}.c22{border-spacing:0;border-collapse:collapse;margin-right:auto}.c23{background-color:#ffffff;max-width:540pt;padding:36pt 36pt 36pt 36pt}.c11{font-size:10pt;font-family:"Courier New";font-weight:400}.c6{color:#000000;font-weight:400;text-decoration:none}.c30{padding:0;margin:0}.c28{margin-left:72pt;padding-left:0pt}.c7{font-size:10pt;font-style:italic}.c8{vertical-align:baseline;font-family:"Arial"}.c1{color:inherit;text-decoration:inherit}.c13{font-weight:700}.c12{height:0pt}.c20{text-indent:36pt}.c18{font-style:italic}.c17{font-size:10pt}.c9{height:11pt}.c27{font-style:normal}.c26{vertical-align:super}.title{padding-top:0pt;color:#000000;font-size:26pt;padding-bottom:3pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}.subtitle{padding-top:0pt;color:#666666;font-size:15pt;padding-bottom:16pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}li{color:#000000;font-size:11pt;font-family:"Arial"}p{margin:0;color:#000000;font-size:11pt;font-family:"Arial"}h1{padding-top:20pt;color:#000000;font-size:20pt;padding-bottom:6pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h2{padding-top:18pt;color:#000000;font-size:16pt;padding-bottom:6pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h3{padding-top:16pt;color:#434343;font-size:14pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h4{padding-top:14pt;color:#666666;font-size:12pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h5{padding-top:12pt;color:#666666;font-size:11pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h6{padding-top:12pt;color:#666666;font-size:11pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;font-style:italic;orphans:2;widows:2;text-align:left}</style></head><body class="c23"><p class="c4"><span class="c16 c8 c13">Title:</span></p><p class="c4"><span class="c5 c8">Guiding the refinement of biochemical knowledgebases with ensembles of metabolic networks and semi-supervised learning</span></p><p class="c4 c9"><span class="c5 c8"></span></p><p class="c4"><span class="c16 c8 c13">Authors:</span></p><p class="c4"><span class="c13">Gregory L. Medlock,</span><span class="c13 c26">1</span><span class="c13">&nbsp;Jason A. Papin</span><span class="c13 c26">1,2,3,</span><span class="c16 c8 c13">*</span></p><p class="c4"><span class="c26">1</span><span class="c5 c8">Department of Biomedical Engineering, University of Virginia, Charlottesville, VA, USA</span></p><p class="c4"><span class="c26">2</span><span class="c5 c8">Department of Medicine, Division of Infectious Diseases &amp; International Health, University of Virginia, Charlottesville, VA, USA</span></p><p class="c4"><span class="c26">3</span><span class="c5 c8">Department of Biochemistry &amp; Molecular Genetics, University of Virginia, Charlottesville, VA, USA</span></p><p class="c4"><span>*Correspondence: papin@virginia.edu</span></p><hr style="page-break-before:always;display:none;"><p class="c4 c9"><span class="c5 c8"></span></p><p class="c4"><span class="c13">Abstract</span></p><p class="c4"><span class="c5 c8">Mechanistic models are becoming common in biology and medicine. These models are often more generalizable than data-driven models because they explicitly represent known interactions between components. While this generalizability has advantages, it also creates a dilemma: how should efforts be focused to improve model performance? We present a semi-supervised machine learning approach to this problem and apply it to genome-scale metabolic network reconstructions. We generate an ensemble of candidate models consistent with experimental data, then perform in silico simulations for which improved predictiveness is desired. We apply semi-supervised learning to these simulation results to identify structural variation in ensemble members that maximally influences variance in simulation outcomes across the ensemble. These structural variants are high priority candidates for curation through targeted experimentation. We demonstrate this approach by applying it to 29 bacterial species to identify curation targets that improve gene essentiality predictions. Our approach represents a fully automated, scalable, and performance-driven recommendation system for curating hypothesis-driven models.</span></p><p class="c4 c9"><span class="c5 c8"></span></p><p class="c4"><span class="c8 c13 c16">Significance</span></p><p class="c4"><span class="c5 c8">Mechanistic computational models, such as metabolic and signaling networks, are becoming common in biology. These models contain a comprehensive representation of components and interactions for a given system, making them widely applicable and often more predictive than simpler models. However, their size and connectivity makes it difficult to identify which parts of a model need to be changed to improve performance further. We developed a strategy to guide this process and applied it to metabolic models for a set of bacterial species. We use it to identify model components that should be investigated, and show that it improves predictive performance. This approach can systematically aide the curation of metabolic models without relying on the intuition of the curator.</span></p><hr style="page-break-before:always;display:none;"><p class="c4 c9"><span class="c5 c8"></span></p><p class="c4"><span class="c13">Introduction</span></p><p class="c4"><span>Genome-scale metabolic network reconstructions (GENREs) are knowledgebases describing metabolic capabilities and their biochemical basis for entire organisms. GENREs can be mathematically formalized and </span><span>combined with numerical representations of biological constraints</span><span>&nbsp;and objectives to create genome-scale metabolic models (GEMs). These GEMs can be used to predict biological outcomes (e.g. gene essentiality, growth rate) given an environmental context (e.g. metabolite availability) </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/mufrM&amp;sa=D&amp;ust=1545253714959000">(1)</a></span><span>.</span><span>&nbsp;GEMs are now used widely for well-studied organisms such as </span><span class="c18">Escherichia coli</span><span>&nbsp;and</span><span>&nbsp;</span><span class="c18">Saccharomyces cerevisiae</span><span>,</span><span>&nbsp;but GEMs for most other organisms are much more taxing to create and curate, partially due to the exhaustive and manually-driven steps required </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/G7As6&amp;sa=D&amp;ust=1545253714960000">(2)</a></span><span>.</span></p><p class="c3"><span>Systems for automatically generating GEMs of sufficient quality for limited purposes have been developed </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/87dmN&amp;sa=D&amp;ust=1545253714961000">(3)</a></span><span>,</span><span class="c5 c8">&nbsp;but the methods used to further curate GEMs are nearly universally under-reported in the literature. Curation methods for GEMs that take researchers many months to years to develop are often summarized qualitatively with limited description. This is not surprising, given the difficulty in prioritizing areas for curation of network-based, highly connected mechanistic models such as GEMs.</span></p><p class="c3"><span>In practice, h</span><span>euristics are typically used to prioritize curation, such as curating portions of the GEM directly involved in the manipulation of a metabolite, gene, or pathway of known interest. These heuristics, combined with targeted literature searches, allow task-based curation and GEM evaluation, which is increasingly supported in software related to genome-scale metabolic modeling </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/xOLM7%2BwDlXZ&amp;sa=D&amp;ust=1545253714961000">(4, 5)</a></span><span>.</span><span class="c5 c8">&nbsp;However, identifying the network components that influence the predictions of interest is not an intuitive process because biological networks are generally highly connected.</span></p><p class="c3"><span>One way to view this issue is through the lens of a sensitivity analysis, asking how much variation in the parameters of a model will impact a simulation of interest. Such an approach has been developed and applied to dynamic models of biological networks </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/t2JUC&amp;sa=D&amp;ust=1545253714962000">(6)</a></span><span>,</span><span>&nbsp;which relies on quantified uncertainty in the structure of a model. Uncertainty quantification has been applied at the level of individual components within a GEM, either by considering the probability of a function being present in a network based on sequence comparisons </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/lZgxl&amp;sa=D&amp;ust=1545253714962000">(7)</a></span><span>&nbsp;or by leveraging network structure to more accurately estimate these probabilities </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/xfXfF&amp;sa=D&amp;ust=1545253714962000">(8)</a></span><span>.</span><span class="c5 c8">&nbsp;However, an approach that unifies a probabilistic views of GEM structure with simulations performed with them, which would enable structural sensitivity analysis for GEMs, has not been developed to our knowledge. </span></p><p class="c3"><span>Recently, we developed a framework for the generation of ensembles of GEMs which can be applied to improve predictive performance over that of an individual GEM </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/WQRJO&amp;sa=D&amp;ust=1545253714963000">(9)</a></span><span>.</span><span>&nbsp;This approach is analogous to the use of ensembles of data-driven models </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/m1GF6&amp;sa=D&amp;ust=1545253714963000">(10)</a></span><span>&nbsp;or hypothesis-driven models such as signaling networks </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/DKZdJ&amp;sa=D&amp;ust=1545253714963000">(11)</a></span><span>, and has been applied to metabolic networks for dynamic modeling as well </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/kA6tI&amp;sa=D&amp;ust=1545253714964000">(12)</a></span><span>. </span><span>Here, we extend our framework to prioritize</span><span>&nbsp;curation of network-based biological models by coupling ensemble simulations with semi-supervised learning.</span><span class="c5 c8">&nbsp;One of the central tenets of systems biology is that models represent our hypotheses about how an organism functions. As such, we can use these models to simulate the behavior we expect according to our hypotheses. Our approach takes advantage of this, generating many hypotheses (e.g. an ensemble) and coupling them with machine learning to identify experiments that optimally improve our understanding of an organism.</span></p><p class="c3 c9"><span class="c5 c8"></span></p><p class="c4"><span class="c16 c8 c13">Results</span></p><p class="c3"><span>Our approach, which can be generalized to any mechanistic model and simulation task of interest, is summarized as follows. First, we generate many models that are each consistent with experimental data, forming an ensemble of models (</span><span class="c13">Figure 1a</span><span>). We then perform a set of simulations using the ensemble that are related to a task of interest, such as drug target identification or production of a metabolite of commercial interest (</span><span class="c13">Figure 1b</span><span>). Using the output of these simulations, we perform unsupervised learning to generate phenotypic </span><span>clusters</span><span>&nbsp;of models, where clustering is determined by similarity of simulation profiles across the entire set of simulations (</span><span class="c13">Figure 1c</span><span>). We then apply supervised learning to predict simulation cluster membership for each model using the values of variable parameters in that model as input (</span><span class="c13">Figure 1c</span><span>). The relative importance of these model parameters in the supervised learning model indicates the impact that uncertainty in that parameter has on simulation outcomes across the ensemble (</span><span class="c13">Figure 1d</span><span>). In other words, resolving the true state of these parameters will maximally reduce uncertainty in the simulations performed with the ensemble. </span><span>Here,</span><span>&nbsp;we apply this approach to the task of reducing uncertainty in predicted gene essentiality for 31 bacterial species (</span><span class="c13">Figure 1a-d</span><span>). </span><span>We generate an ensemble for each species using previously published growth phenotyping data </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/beLlp&amp;sa=D&amp;ust=1545253714966000">(13)</a></span><span>,</span><span>&nbsp;predict the effect of genome-wide single gene knockouts, then apply semi-supervised learning as described above</span><span>.</span></p><a id="t.235357dbd9a32737fbec6e2a36ff8cb56f265ad9"></a><a id="t.0"></a><table class="c22"><tbody><tr class="c12"><td class="c21" colspan="1" rowspan="1"><p class="c25"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 706.00px; height: 401.33px;"><img alt="" src="images/image16.png" style="width: 706.00px; height: 401.33px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p></td></tr><tr class="c12"><td class="c21" colspan="1" rowspan="1"><p class="c25"><span class="c13 c17">Figure 1.</span><span class="c17">&nbsp;Approach for a simulation-guided recommendation system for curation of genome-scale metabolic models. </span><span class="c13 c17">a)</span><span class="c17">&nbsp;A draft GEM is generated using ModelSEED. Algorithmic gapfilling is applied so that the GEM can recapitulate experimental observations, and the process is repeated to identify alternative solutions. These alternative solutions are assembled into an ensemble of GEMs, each of which contains the original content of the draft GEM and a unique set of gapfilled reactions. </span><span class="c13 c17">b)</span><span class="c17">&nbsp;Single gene knockouts are performed using the ensemble, in which production of biomass is evaluated when reactions requiring each gene are inactivated. </span><span class="c13 c17">c-d)</span><span class="c6 c8 c17 c27">&nbsp;Semi-supervised approach for identifying curation targets based on ensemble simulations. Unsupervised machine learning is applied to the ensemble simulation results, generating two simulation clusters (cluster 1 [red] and cluster 2 [blue]). Here, principal coordinate analysis is used to visualize the similarity of simulation profiles for all models within an ensemble. Simulation clusters are then used as labels in supervised machine learning, which are predicted using model reaction content as input to a random forest classifier. Curation is prioritized based on the features contributing to classifier performance.</span></p></td></tr></tbody></table><p class="c4 c9"><span class="c5 c8"></span></p><p class="c3"><span>Given our objective of identifying the most impactful experiment or curation effort to improve the quality of our models, we required that our ensembles were large enough to saturate the space of unique simulation results (i.e. predicted behavior) and model structures (i.e. hypotheses). Toward this end, we first subsampled the ensemble for each species and determined the unique reaction content within each subsample (</span><span class="c13">Figure 2a</span><span>). We found that unique reaction content </span><span>plateaued</span><span>&nbsp;or nearly plateaued with ensembles </span><span>containing </span><span>as few as </span><span>100</span><span>-200</span><span>&nbsp;</span><span>models</span><span class="c5 c8">, suggesting the ensembles we generated sufficiently saturate the space of unique reactions. In other applications, we suspect that organisms with lower quality GEMs (e.g. more gaps in their metabolic network) or less phenotypic profiling data may require additional sampling to saturate this space.</span></p><p class="c3"><span>For gene essentiality simulations, the number of variable predictions (e.g. number of genes for which at least one ensemble member disagrees with another member) plateaued in a similar manner (</span><span class="c13">Figure 2b</span><span>). We also performed subsampling for predictions of genes essential for biomass production, which exhibited similar properties of convergence (</span><span class="c13">Supplemental Figure 1a-b</span><span>). </span><span>Taken together, these subsampling-based results confirm that ensembles containing 1000 models generated using our reconstruction pipeline sufficiently represent the network structure space (e.g. unique reactions) and prediction space (e.g. essentiality profiles) possible given the input data. This behavior is consistent with our previous work examining the performance of ensembles of GEMs for </span><span class="c18">Pseudomonas aeruginosa</span><span>, in which various aspects of ensemble performance nearly plateaued with only 50 GEMs </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/WQRJO&amp;sa=D&amp;ust=1545253714969000">(9)</a></span><span>. However, in order to ensure that an adequate number of samples are included for downstream semi-supervised machine learning analyses, we maintain the full ensemble of 1000 GEMs for each species in all analyses.</span></p><a id="t.c312fadedfbd7fe3b33f22e00700a1da2660bfda"></a><a id="t.1"></a><table class="c22"><tbody><tr class="c12"><td class="c21" colspan="1" rowspan="1"><p class="c25"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 706.00px; height: 218.67px;"><img alt="" src="images/image18.png" style="width: 706.00px; height: 218.67px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p></td></tr><tr class="c12"><td class="c29" colspan="1" rowspan="1"><p class="c25"><span class="c13 c17">Figure 2.</span><span class="c17">&nbsp;Subsampling of ensemble content and simulations to demonstrate exhaustive sampling of the solution space. </span><span class="c13 c17">a)</span><span class="c17">&nbsp;The variable reaction content in ensemble members as a function of increasing ensemble size. A variable reaction is any reaction that has a variable state across any member of the subsampled group (e.g. it is off in some members and on in some members, but neither entirely off nor entirely on). For each species, the mean number of variable reactions in subsamples of GEMs is shown by the solid line, with the standard deviation shown as light fill of the same color above and below the mean. Subsampling was performed with 1000 draws per subsample size. Ensembles were sampled at intervals of 20 members, e.g. 20, 40, 60&hellip; until reaching the size of the entire ensemble. </span><span class="c13 c17">b)</span><span class="c17">&nbsp;Variability in gene essentiality simulations within subsamples of ensemble members. Using the same subsampling procedure as in panel </span><span class="c13 c17">a</span><span class="c6 c8 c17 c27">, the number of genes with at least one GEM in the subsample with a simulation outcome different than the rest (e.g. non-consensus) was determined. The mean for each subsample size is shown by the solid line, with the standard deviation shown as light fill of the same color above and below the mean.</span></p></td></tr></tbody></table><p class="c4 c9"><span class="c5 c8"></span></p><p class="c4"><span>Each species&rsquo; ensemble contained 19.27</span><span>&nbsp;+/- 8.66 genes</span><span>&nbsp;(mean +/- standard deviation) for which at least one GEM&rsquo;s prediction of essentiality disagreed with another GEM in the ensemble, representing 3.11</span><span>&nbsp;+/- 1.39%</span><span>&nbsp;of total metabolic gene content. For the unsupervised machine learning portion of this analysis, we performed </span><span class="c18">k</span><span>-means clustering on the gene essentiality simulations from </span><span>each species&rsquo; ensemble</span><span>&nbsp;separately. We chose </span><span class="c18">k</span><span>&nbsp;= 2 to generate two clusters for each species, each of which contain GEMs from the ensemble with similar gene essentiality simulation profiles. </span><span>The results are visualized for a subset of species using principal coordinate analysis (PCoA) in </span><span class="c13">Figure 3a</span><span>&nbsp;and </span><span class="c13">3c</span><span>.</span><span>&nbsp;Although we chose </span><span class="c18">k</span><span>&nbsp;= 2 here to illustrate our approach, the separation of models in PCoA space suggests that for many species, determining a larger number of clusters might be advantageous.</span><span>&nbsp;For example, w</span><span>hile </span><span class="c18">k</span><span>&nbsp;= 2 generates two maximally-different simulation clusters, there may be more than two distinct </span><span class="c18">in silico</span><span>&nbsp;phenotypic clusters</span><span>&nbsp;that represent significant differences in hypothesized model behavior. Accounting for the presence of these smaller clusters may identify important network features that would otherwise only be found through multiple iterations of clustering with </span><span class="c18">k</span><span class="c5 c8">&nbsp;= 2 and refinement of the ensemble.</span></p><p class="c3"><span>We next sought to identify the reactions that vary across an ensemble that are associated with membership in each cluster. For this objective, we trained a random forest classifier</span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/HJkaR&amp;sa=D&amp;ust=1545253714973000">(14)</a></span><span>&nbsp;to predict cluster membership for each GEM from its reaction content</span><span>.</span><span>&nbsp;The classifier for every species had an out-of-bag accuracy above </span><span>97%</span><span class="c5 c8">, indicating that gene essentiality cluster membership can easily be predicted from reaction content within the ensembles. To prioritize candidate reactions for curation of each species&rsquo; ensemble, we examined the features that contributed the most to classifier performance.</span></p><p class="c3"><span>One way to resolve the identity of these prioritized reactions is a targeted literature search, which might reveal information that has not been incorporated into genomic or metabolic databases. For example, in the ensemble for </span><span class="c18">Corynebacterium glutamicum</span><span>, the strongest driver of essentiality cluster membership </span><span>was inclusion of the reaction L-threonine acetaldehyde-lyase</span><span>, which converts L-threonine to glycine and acetaldehyde (</span><span class="c13">Figure 3</span><span class="c13">b</span><span>). This activity is known to occur in </span><span class="c18">C. glutamicum</span><span>, but was not </span><span>annotated </span><span>in its genome because the enzyme primarily uses L-serine as a substrate (serine hydroxymethyltransferase, </span><span class="c13">Figure 3b</span><span>), while conversion of L-threonine is considered a side activity </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/cA6YJ%2BV3SZO&amp;sa=D&amp;ust=1545253714974000">(15, 16)</a></span><span>. The correct function was present in 97 of the ensemble members in the blue cluster and 0 members of the red cluster, representing 84.3% of the blue cluster. Given that these 97 models only represent 9.8% of the ensemble for </span><span class="c18">C. glutamicum</span><span>, this function would have been overlooked if ensemble features were chosen by consensus across the ensemble rather than via semi-supervised learning. </span><span>This result highlights the ability of our approach to identify discrepancies between databases and the literature (such as under-annotated promiscuous enzyme activity) while also prioritizing investigation into these issues based on their contribution to prediction uncertainty.</span><span class="c5 c8">&nbsp;</span></p><p class="c3"><span>Our approach is focused on prioritizing curation efforts to reduce uncertainty in model simulations. However, whether this approach generally improves the predictive performance of GEMs is unclear. To investigate this question, we evaluated the performance of a subset of ensembles for which experimental genome-wide gene essentiality datasets derived from </span><span class="c18">in vitro</span><span>&nbsp;growth on a rich medium were available. </span><span>Suitable datasets were identified for </span><span class="c18">Staphylococcus au</span><span class="c18">reus </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/2Olqk&amp;sa=D&amp;ust=1545253714974000">(17)</a></span><span>&nbsp;and </span><span class="c18">Haemophilus influenzae</span><span>&nbsp;</span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/C2qPA&amp;sa=D&amp;ust=1545253714975000">(18)</a></span><span>. E</span><span>ach GEM in the ensembles for these species was evaluated using precision (the ratio of true positives to the sum of true and false positives) and recall (the ratio of true positives to the sum of true positives and false negatives; </span><span class="c13">Figure 3d</span><span>). For both species, ensemble members had variable precision and recall, and simulation cluster membership is associated with a difference in both precision and recall (</span><span class="c18">p</span><span>&nbsp;&lt; 0.0001, Mann-Whitney </span><span class="c18">U</span><span>-test with false discovery rate control via Benjamini Hochberg procedure). </span><span>Based on this result, selecting models for future analyses by excluding one of the two simulation clusters will improve performance as evaluated against experimental data. Thus, our semi-supervised curation guidance approach will improve model performance when the true state of a parameter is associated with the cluster with better performance.</span></p><p class="c3 c9"><span class="c5 c8"></span></p><p class="c4"><span class="c16 c8 c13">Discussion</span></p><p class="c3"><span>The analysis we performed demonstrates just one possible path towards the </span><span>goal</span><span>&nbsp;of </span><span>reducing uncertainty in </span><span>our understanding of biochemical networks within the semi-supervised framework we propose. Changes to the process can be rationalized for new goals; for example, we previously demonstrated that introducing random weights on inclusion of each reaction during algorithmic gapfilling can generate more diverse ensembles </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/WQRJO&amp;sa=D&amp;ust=1545253714976000">(9)</a></span><span>. If none of the ensemble members generated by our pipeline adequately represented metabolism for an organism (e.g. their gene essentiality simulation results were vastly different than experimental observations), we could introduce such random variance to increase the likelihood of generating some ensemble members that reflect biological reality. Such an approach may be necessary for organisms with metabolic repertoires differing substantially from those represented in popular biochemical databases (e.g. gut microbes, intracellular parasites). Inclusion of methods for proposing novel hypothetical enzymatic function could complement our approach for such organisms </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/goaRi%2BLPxAy&amp;sa=D&amp;ust=1545253714976000">(19, 20)</a></span><span class="c5 c8">.</span></p><p class="c3"><span>Our approach builds on work in other disciplines in which uncertainty quantification and reduction are applied to understand or improve the behavior of domain-specific models. For example, in petroleum engineering, a similar ensemble-based approach is used to derive value of information (VOI) estimates for resolving parameter values in models of oil reservoir management </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/PjDgh&amp;sa=D&amp;ust=1545253714976000">(21)</a></span><span>. In the application of our framework demonstrated here, we effectively derive VOI estimates for resolving reaction presence or absence. Taking a VOI approach for biological discovery and to improve models used in various facets of biotechnology could help automate workflows and substantially reduce costs by prioritizing experiments. As the diversity and depth of organisms that mechanistic models such as GEMs are being constructed for increases, such approaches will be vital to continue to improve their quality and predictiveness </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/xU60c%2B5DSzt&amp;sa=D&amp;ust=1545253714977000">(22, 23)</a></span><span class="c5 c8">.</span></p><p class="c3 c9"><span class="c5 c8"></span></p><a id="t.d16ea2e43c6444e3f2f975dd82a3d0bc04051a1b"></a><a id="t.2"></a><table class="c22"><tbody><tr class="c12"><td class="c21" colspan="1" rowspan="1"><p class="c25"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 706.00px; height: 532.00px;"><img alt="" src="images/image17.png" style="width: 706.00px; height: 532.00px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p></td></tr><tr class="c12"><td class="c21" colspan="1" rowspan="1"><p class="c25"><span class="c13 c17">Figure 3.</span><span class="c17">&nbsp;Application of the semi-supervised curation guidance pipeline to bacterial species. </span><span class="c13 c17">a &amp; c</span><span class="c13 c17">)</span><span class="c17">&nbsp;Ensemble gene essentiality simulations and unsupervised learning for three representative species. Each heatmap displays essentiality simulations for each ensemble member (row label) after inactivating a single gene (column label). </span><span class="c17">Black</span><span class="c17">&nbsp;cells indicate genes predicted to be essential while white cells</span><span class="c17">&nbsp;indicate genes predicted to be non-essential.</span><span class="c17">&nbsp;Principal coordinate analysis (PCoA) plots next to each heatmap show the similarity between gene essentiality simulation profiles for each ensemble member. Within each PCoA plot, each point represents an ensemble member, colored by cluster membership as determined with </span><span class="c7">k</span><span class="c17">-means clustering </span><span class="c17">(</span><span class="c7">k</span><span class="c17">=2)</span><span class="c17">. Percent variance in the pairwise distance matrix explained by each principal coordinate is indicated in parentheses. See Methods for analysis details. </span><span class="c13 c17">b</span><span class="c13 c17">)</span><span class="c17">&nbsp;</span><span class="c17">Example of a network feature driving simulation cluster membership</span><span class="c17">. The top reaction, catalyzed by serine hydroxymethylt</span><span class="c17">ransferase, is the primary activity of the enzyme, which was annotated in the draft GEM. The bottom reaction is a promiscuous function of the same enzyme, which was the most important reaction in determining cluster membership </span><span class="c17">(</span><span class="c7">k</span><span class="c17">=2)</span><span class="c17">&nbsp;for </span><span class="c7">Corynebacterium glutamicum</span><span class="c17">. (abbreviations; THF is tetrahydrofolate and 5,10-mTHF is 5,10-Methylenetetrahydrofolate). Water is excluded from reactants in the top reaction to simplify visualization. </span><span class="c13 c17">d)</span><span class="c17">&nbsp;Evaluation of performance of GEMs in </span><span class="c17">each simulation cluster</span><span class="c17">&nbsp;compared to genome-wide gene essentiality data. Essentiality datasets from </span><span class="c7">in vitro</span><span class="c17">&nbsp;experiments were collected for </span><span class="c7">Haemophilus influenzae</span><span class="c17">&nbsp;and </span><span class="c7">Staphylococcus aureus</span><span class="c17">. Precision (TP/[TP + FP], TP = true positives, FP = false positives) and recall (TP/[TP+FN], FN = false negatives) were calculated for each ensemble member for each species. Small red and blue circles indicate an individual ensemble member, colored by simulation cluster membership. Large red and blue circles indicate mean behavior for ensemble members from each cluster, and error bars of same color extend above and below the mean by one standard </span><span class="c17">deviation</span><span class="c17">.</span></p></td></tr></tbody></table><p class="c4 c9"><span class="c5 c8"></span></p><p class="c4 c9"><span class="c16 c8 c13"></span></p><p class="c4"><span class="c13">Methods</span></p><p class="c4"><span class="c18">Organism selection</span></p><p class="c4"><span>Organisms with available growth phenotype data were extracted from Plata, </span><span class="c18">et al.</span><span>&nbsp;</span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/beLlp&amp;sa=D&amp;ust=1545253714981000">(13)</a></span><span>.</span><span>&nbsp;To identify a representative genome for each species, we queried the PATRIC database </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/rGwAW&amp;sa=D&amp;ust=1545253714981000">(24)</a></span><span class="c5 c8">&nbsp;with the genus and species name for all organisms in the study, then selected a single genome from PATRIC based on decision criteria described as follows. When a reference genome was assigned for the species, the genome identifier for the reference genome was chosen. If no reference genome was available, a genome listed as &ldquo;representative&rdquo; was chosen. When multiple genomes with the &ldquo;representative&rdquo; status were available, we chose the first genome listed. If a selected representative genome contained more than 10 contigs, a representative genome with fewer contigs was chosen. Strain identifiers were not provided in the study from which data was drawn, so these selection criteria were developed to select the highest-quality genome available for the species in the study. Selected genome identifiers are available in Supplemental Table S1.</span></p><p class="c4"><span>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Organism selection was further refined by only including those from Plata, </span><span class="c18">et al</span><span>. </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/beLlp&amp;sa=D&amp;ust=1545253714982000">(13)</a></span><span>&nbsp;which grew in at least 10 of the single-carbon source Biolog conditions. The experimental growth threshold originally used in the paper from which data were drawn was used (&gt;10 colorimetric units of tetrazolium dye reduction; originally scaled between 0 and 100 based on positive [100 units] and negative [0 units] controls). This choice was made with the recognition that the tetrazolium dye measures redox activity and not actual biomass production; for the purpose of our study, we assume that detectable redox activity above 10 relative units would require biomass production. After this initial selection step, </span><span class="c18">Brachybacterium faecium</span><span>&nbsp;and </span><span class="c18">Gordonia bronchialis</span><span>&nbsp;were also removed from the analysis because no solutions existed to enable biomass production using the universal reaction bag for either species. </span><span class="c18">Bacillus megaterium</span><span>&nbsp;was excluded because only one gapfill solution was found across all gapfilling cycles</span><span>. Similarly, </span><span class="c18">Stenotrophomonas maltophilia</span><span>&nbsp;was excluded because only two unique gapfill solutions were fo</span><span>und.</span><span>&nbsp;In total, the full analysis pipeline was applied to </span><span>29</span><span class="c5 c8">&nbsp;species.</span></p><p class="c4 c9"><span class="c5 c8"></span></p><p class="c4"><span class="c2 c8">Generation of draft genome-scale metabolic models</span></p><p class="c4"><span>Draft-quality genome-scale metabolic models (GEMs) were generated using the ModelSEED reconstruction pipeline </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/87dmN&amp;sa=D&amp;ust=1545253714983000">(3)</a></span><span>&nbsp;accessed through PATRIC in August, 2018 </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/rGwAW&amp;sa=D&amp;ust=1545253714983000">(24)</a></span><span>.</span><span>&nbsp;PATRIC servers were queried to generate GEMs formatted for use in cobrapy </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/ICKNq&amp;sa=D&amp;ust=1545253714983000">(25)</a></span><span>&nbsp;using the Mackinac package </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/OFgEz&amp;sa=D&amp;ust=1545253714984000">(26)</a></span><span>.</span></p><p class="c4 c9"><span class="c5 c8"></span></p><p class="c4"><span class="c18">Representative media</span></p><p class="c4"><span>The base medium for biolog conditions was derived from the ModelSEED media compositions for biolog plates. Flux variability analysis was used to identify metabolites which had essential uptake reactions in all complete media-gapfilled reconstructions from PATRIC. Based on this analysis, we added Heme and H2SO3 to the base biolog composition used </span><span class="c18">in silico</span><span class="c5 c8">&nbsp;(i.e. uptake of heme and H2SO3 was allowed in all conditions). For each single carbon source, appropriate identifiers were found in the ModelSEED database. For metabolites with ambiguous chemical identities (e.g. metabolites that Biolog does not provide isomer composition for, such as D-Galactose), only one isomer was selected from ModelSEED to represent the condition. Carbon sources that are complex mixtures of metabolites (gelatin) or polymers (pectin) were excluded from analyses.</span></p><p class="c4 c9"><span class="c5 c8"></span></p><p class="c4"><span class="c18">Algorithmic gapfilling</span></p><p class="c4"><span>Each individual gapfilling step, corresponding to enabling biomass production on a single media source, was performed using the following algorithm adapted from our previous work </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/WQRJO&amp;sa=D&amp;ust=1545253714985000">(9)</a></span><span>.</span><span>&nbsp;This algorithm is in essence the same as parsimonious flux balance analysis (pFBA, </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/sngqz&amp;sa=D&amp;ust=1545253714985000">(27)</a></span><span>),</span><span class="c5 c8">&nbsp;except that the parsimonious assumption of minimization of the sum of all fluxes is only applied to reactions from a universal reaction bag that are activated to allow flux through the network.</span></p><p class="c19 c9"><span class="c5 c8"></span></p><p class="c19"><span class="c5 c8">Algorithm 1: pFBA-based gapfilling</span></p><p class="c19 c9"><span class="c5 c8"></span></p><p class="c19"><span>Min </span><img src="images/image1.png"><span class="c5 c8">, subject to:</span></p><p class="c19 c20"><img src="images/image2.png"></p><p class="c19 c20"><img src="images/image3.png"></p><p class="c19 c20"><img src="images/image4.png"></p><p class="c19 c20"><img src="images/image5.png"></p><p class="c19 c20 c9"><span class="c5 c8"></span></p><p class="c19"><span>Where </span><img src="images/image6.png"><span>&nbsp;is the stoichiometric matrix representing the model to be gapfilled, </span><img src="images/image7.png"><span>&nbsp;is the vector of fluxes through reactions in </span><img src="images/image6.png"><span>, </span><img src="images/image8.png"><span>&nbsp;is the stoichiometric matrix representing the reaction database from which reactions are activated to fill gaps, </span><img src="images/image9.png"><span>&nbsp;is the vector of fluxes through reactions in </span><img src="images/image8.png"><span>, </span><img src="images/image10.png"><span>&nbsp;is flux through the biomass reaction, </span><img src="images/image11.png"><span>&nbsp;and </span><img src="images/image12.png"><span>&nbsp;are lower and upper bounds of flux through reactions in the original model, respectively, &nbsp;and </span><img src="images/image13.png"><span>&nbsp;and </span><img src="images/image14.png"><span>&nbsp;are lower and upper bounds of flux through reactions in the reaction database </span><img src="images/image8.png"><span class="c5 c8">.</span></p><p class="c19 c20"><span>The formulation is identical to the original formulation of pFBA, except for four key differences. First, we only require an arbitrarily low amount of flux through biomass, rather than the maximum amount of biomass, meant to represent a binary growth condition. Second, we introduce a universal reaction bag (</span><img src="images/image8.png"><span>) and associated flux variables for each reaction in U (</span><img src="images/image9.png"><span>). Third, only fluxes through reactions in </span><img src="images/image8.png"><span>&nbsp;are penalized; fluxes through reactions in the model being gapfilled (</span><img src="images/image6.png"><span>) are not penalized. Fourth, rather than explicitly splitting all reactions into irreversible reactions, we take advantage of solver-level interfaces implemented in cobrapy through the optlang package </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/37ibn&amp;sa=D&amp;ust=1545253714992000">(28)</a></span><span>&nbsp;that allow introduction of absolute values into the objective (this is done out of convenience in our implementation; this aspect of the problem formulation is identical to the same aspect in pFBA at the solver level) </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/37ibn&amp;sa=D&amp;ust=1545253714993000">(28)</a></span><span>.</span><span>&nbsp;As in Biggs et al. </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/WQRJO&amp;sa=D&amp;ust=1545253714993000">(9)</a></span><span>,</span><span class="c5 c8">&nbsp;the solution to this optimization problem activates reactions in the universal reaction bag with the minimum sum of fluxes necessary to enable flux through the biomass reaction in a given condition.</span></p><p class="c19 c9"><span class="c5 c8"></span></p><p class="c19"><span class="c18">Generating ensembles from gapfill solutions</span></p><p class="c19"><span class="c5 c8">For each organism, the entire algorithm for generating an ensemble is as follows:</span></p><p class="c19 c9"><span class="c5 c8"></span></p><p class="c19"><span class="c0">For i in number_of_ensemble_members:</span></p><p class="c19"><span class="c0">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Randomly order a selection of J media conditions</span></p><p class="c19 c20"><span class="c0">For a single condition j in J:</span></p><ol class="c30 lst-kix_q1ktesomj3tk-0 start" start="1"><li class="c19 c28"><span class="c0">Set model bounds to represent media condition</span></li><li class="c19 c28"><span class="c0">Optimize using pFBA-based gapfilling</span></li><li class="c19 c28"><span class="c0">Add activated reactions/remove flux </span></li></ol><p class="c19"><span class="c0">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; minimization penalty for those reactions</span></p><p class="c19 c20"><span class="c0">Store solution from this iteration</span></p><p class="c19"><span class="c0">Create an ensemble where each member contains the set of </span></p><p class="c19 c20"><span class="c0">reactions added over an iteration through all media conditions.</span></p><p class="c19"><span class="c5 c8">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></p><p class="c4"><span>We performed this procedure for 1000 cycles for each species (i.e. </span><span class="c11">number_of_ensemble_members = 1000</span><span>). All species included in the study grew </span><span>in at least 10</span><span>&nbsp;</span><span class="c18">in vitro</span><span>&nbsp;single carbon source media conditions (i.e. </span><span class="c11">J</span><span>&nbsp;contained at least 10 conditions); for each species, all positive growth conditions were used to gapfill during each cycle. After removing duplicate gapfill solutions, all species included for further analyses had 970-1000 members in their ensemble (species not considered after this point are detailed in </span><span class="c18">Organism selection</span><span class="c5 c8">).</span></p><p class="c4 c9"><span class="c5 c8"></span></p><p class="c4"><span class="c2 c8">Ensemble Flux Balance Analysis and Ensemble Gene Essentiality</span></p><p class="c4"><span>Ensemble flux balance analysis and ensemble gene essentiality screens were performed using Medusa </span><span>v</span><span>0.1.2 (unpublished, https://github.com/gregmedlock/Medusa) and </span><span>cobrapy </span><span>v0.13 </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/ICKNq&amp;sa=D&amp;ust=1545253714997000">(25)</a></span><span>.</span><span class="c5 c8">&nbsp;The GNU linear programming kit (GLPK) was used as the numerical solver in all cases. For all simulations, rich medium was used (1,000 mmol/gram dry weight*hr uptake allowed for all metabolites with a transport reaction; commonly referred to as &ldquo;complete medium&rdquo;). An arbitrarily low cutoff for flux through biomass in gene essentiality screens was used (1E-6 units of biomass/hr), but varying this quantity between 1E-10 and 1E-3 did not substantially affect essentiality results.</span></p><p class="c4 c9"><span class="c5 c8"></span></p><p class="c4"><span class="c18">Subsampling of ensemble features and predictions</span></p><p class="c4"><span class="c5 c8">For all subsampling performed, 1,000 random draws were made with replacement at each subsample ensemble size. Ensemble sizes for each subsampled population ranged from 20 to 1,000, with subsampling performed in intervals of 20 members (i.e. 20, 40 , 60 &hellip; 1,000 members). When the subsample size exceeded the actual ensemble size (e.g. some species had slightly less than 1,000 members), all ensemble members were subsampled.</span></p><p class="c4 c9"><span class="c5 c8"></span></p><p class="c4"><span class="c18">Clustering of ensemble gene essentiality predictions and prediction of clusters</span></p><p class="c4"><span>Ensemble gene essentiality predictions were clustered into two clusters using </span><span class="c18">k</span><span>-means clustering as implemented in the KMeans class of scikit-learn v0.19.2 </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/v1H8r&amp;sa=D&amp;ust=1545253714999000">(29)</a></span><span>&nbsp;(max iterations=300, convergence tolerance=1E-4, Elkan&rsquo;s</span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/2g437&amp;sa=D&amp;ust=1545253714999000">(30)</a></span><span>&nbsp;algorithm). Gene essentiality predictions were converted to binary data (essential or nonessential) using a cutoff of flux through biomass of 1E-6 mmol/(gDW*hr). Random forest classification was performed to predict cluster membership using </span><span>active features in each ensemble member</span><span>&nbsp;(e.g. presence or absence of a reaction was assigned as True or False in the input, respectively) </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/HJkaR&amp;sa=D&amp;ust=1545253714999000">(14)</a></span><span>.</span><span>&nbsp;The RandomForestClassifier class from scikit-learn v0.19.2 was used (500 trees, quality of splits determined with the gini criterion, no max depth, minimum of 2 samples per split, minimum of 1 sample per leaf, sqrt(number of features) searched at each split, training samples determined for each tree via bootstrap selection with replacement). The default metric in scikit-learn&rsquo;s RandomForestClassifier for determining feature importance, the mean decrease in node purity, was used to calculate feature importance in this study </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/8d4f6&amp;sa=D&amp;ust=1545253715000000">(31)</a></span><span>.</span></p><p class="c4 c9"><span class="c5 c8"></span></p><p class="c4"><span class="c18">Visualization of gene essentiality clusters</span></p><p class="c4"><span>Principal coordinate analysis (PCoA) </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/VWwiV&amp;sa=D&amp;ust=1545253715001000">(32)</a></span><span>&nbsp;was used to visualize ensemble gene essentiality results. PCoA as implemented in scikit-bio v0.5.4 (https://github.com/biocore/scikit-bio) was performed using the hamming distance </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/YxHco&amp;sa=D&amp;ust=1545253715001000">(33)</a></span><span>&nbsp;to compute the pairwise distance matrix.</span></p><p class="c4 c9"><span class="c5 c8"></span></p><p class="c4"><span class="c18">Gene essentiality datasets</span></p><p class="c4"><span>Gene essentiality datasets were identified for species in this study from the </span><span>Online Database of Gene Essentiality (OGEE, </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/XRGIp&amp;sa=D&amp;ust=1545253715002000">(34)</a></span><span>).</span><span>&nbsp;In cases where multiple datasets were available for a given species, the dataset generated using the same strain of the species selected for GENRE reconstruction was selected. If multiple datasets still existed for a species, a single dataset was chosen based on media richness (e.g. more complex media were selected over simpler media). We excluded the essentiality dataset for </span><span class="c18">Streptococcus pneumoniae</span><span>&nbsp;because the total set of screened genes was not included </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/mxooS&amp;sa=D&amp;ust=1545253715002000">(35)</a></span><span>.</span><span>&nbsp;In brief, the authors developed a kanamycin insertion cassette targeted for 693 genes that were selected based having &gt;40% amino acid sequence identity with a set of well-studied organisms. The authors reported the identity of only the essential genes, so non-essential genes that would be in the dataset could not be included in our set of predictions. Based on these selection criteria and limitations, we selected datasets from OGEE for </span><span class="c18">Staphylococcus aureus </span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/2Olqk&amp;sa=D&amp;ust=1545253715003000">(17)</a></span><span>&nbsp;and </span><span class="c18">Haemophilus influenzae</span><span>&nbsp;</span><span class="c6"><a class="c1" href="https://www.google.com/url?q=https://paperpile.com/c/7TZ51d/C2qPA&amp;sa=D&amp;ust=1545253715003000">(18)</a></span><span>.</span></p><p class="c4 c9"><span class="c5 c8"></span></p><p class="c4"><span class="c18">Data and analysis availability</span></p><p class="c4"><span>All data, analysis scripts, results, and models generated are available at </span><span class="c32"><a class="c1" href="https://www.google.com/url?q=https://github.com/gregmedlock/ssl_ensembles&amp;sa=D&amp;ust=1545253715005000">https://github.com/gregmedlock/ssl_ensembles</a></span><span class="c5 c8">&nbsp;and will be archived on Zenodo upon acceptance for publication after peer review. Archived results include the semi-supervised analysis pipeline results for all species in the study (rather than only the representative examples shown here).</span></p><p class="c4 c9"><span class="c5 c8"></span></p><p class="c4"><span class="c16 c8 c13">Acknowledgements</span></p><p class="c24"><span class="c5 c8">We acknowledge funding from National Institutes of Health R01GM108501, T32LM012416, the Thomas F. and Kate Miller Jeffress Memorial Trust, and a Wagner predoctoral fellowship to GLM. We thank Matthew Biggs for thoughtful discussion related to the manuscript and Maureen Carey for helpful comments on drafts.</span></p><p class="c4 c9"><span class="c16 c8 c13"></span></p><p class="c4"><span class="c16 c8 c13">Author Contributions</span></p><p class="c4"><span class="c5 c8">Conceptualization, G.L.M and J.P; Data Curation, G.L.M; Formal Analysis, G.L.M; Investigation, G.L.M; Methodology, G.L.M; Software, G.L.M; Validation, G.L.M; Visualization, G.L.M; Writing - original draft, G.L.M; Writing - Review &amp; Editing, G.L.M and J.P; Funding Acquisition, G.L.M and J.P; Project Administration, J.P; Resources, J.P; Supervision, J.P.</span></p><p class="c4 c9"><span class="c16 c8 c13"></span></p><p class="c4"><span class="c16 c8 c13">Competing Interests Statement</span></p><p class="c4"><span class="c5 c8">The University of Virginia has filed a U.S. Provisional Patent Application (No. 62/744,393) related to this manuscript which describes a curation guidance system for biological network models of which G.L.M and J.P. are inventors.</span></p><p class="c4 c9"><span class="c5 c8"></span></p><p class="c4"><span class="c13">References</span></p><p class="c31"><span class="c14">1. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/mufrM&amp;sa=D&amp;ust=1545253715007000">Oberhardt MA, Palsson B&Oslash;, Papin JA (2009) Applications of genome&#8208;scale metabolic reconstructions. </a></span><span class="c2"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/mufrM&amp;sa=D&amp;ust=1545253715007000">Mol Syst Biol</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/mufrM&amp;sa=D&amp;ust=1545253715007000">&nbsp;5(1):320.</a></span></p><p class="c15"><span class="c14">2. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/G7As6&amp;sa=D&amp;ust=1545253715008000">Thiele I, Palsson B&Oslash; (2010) A protocol for generating a high-quality genome-scale metabolic reconstruction. </a></span><span class="c2"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/G7As6&amp;sa=D&amp;ust=1545253715008000">Nat Protoc</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/G7As6&amp;sa=D&amp;ust=1545253715008000">&nbsp;5(1):93&ndash;121.</a></span></p><p class="c15"><span class="c14">3. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/87dmN&amp;sa=D&amp;ust=1545253715009000">Henry CS, et al. (2010) High-throughput generation, optimization and analysis of genome-scale metabolic models. </a></span><span class="c2"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/87dmN&amp;sa=D&amp;ust=1545253715009000">Nat Biotechnol</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/87dmN&amp;sa=D&amp;ust=1545253715009000">&nbsp;28(9):977&ndash;982.</a></span></p><p class="c15"><span class="c14">4. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/xOLM7&amp;sa=D&amp;ust=1545253715010000">Lieven C, et al. (2018) Memote: A community-driven effort towards a standardized genome-scale metabolic model test suite. </a></span><span class="c2"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/xOLM7&amp;sa=D&amp;ust=1545253715010000">bioRxiv</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/xOLM7&amp;sa=D&amp;ust=1545253715010000">:350991.</a></span></p><p class="c15"><span class="c14">5. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/wDlXZ&amp;sa=D&amp;ust=1545253715011000">Wang H, et al. (2018) RAVEN 2.0: A versatile toolbox for metabolic network reconstruction and a case study on Streptomyces coelicolor. </a></span><span class="c2"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/wDlXZ&amp;sa=D&amp;ust=1545253715011000">PLoS Comput Biol</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/wDlXZ&amp;sa=D&amp;ust=1545253715011000">&nbsp;14(10):e1006541.</a></span></p><p class="c15"><span class="c14">6. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/t2JUC&amp;sa=D&amp;ust=1545253715012000">Babtie AC, Kirk P, Stumpf MPH (2014) Topological sensitivity analysis for systems biology. </a></span><span class="c2"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/t2JUC&amp;sa=D&amp;ust=1545253715012000">Proc Natl Acad Sci U S A</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/t2JUC&amp;sa=D&amp;ust=1545253715012000">&nbsp;111(52):18507&ndash;18512.</a></span></p><p class="c15"><span class="c14">7. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/lZgxl&amp;sa=D&amp;ust=1545253715012000">Benedict MN, Mundy MB, Henry CS, Chia N, Price ND (2014) Likelihood-based gene annotations for gap filling and quality assessment in genome-scale metabolic models. </a></span><span class="c2"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/lZgxl&amp;sa=D&amp;ust=1545253715012000">PLoS Comput Biol</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/lZgxl&amp;sa=D&amp;ust=1545253715013000">&nbsp;10(10):e1003882.</a></span></p><p class="c15"><span class="c14">8. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/xfXfF&amp;sa=D&amp;ust=1545253715013000">Plata G, Fuhrer T, Hsiao T-L, Sauer U, Vitkup D (2012) Global probabilistic annotation of metabolic networks enables enzyme discovery. </a></span><span class="c2"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/xfXfF&amp;sa=D&amp;ust=1545253715013000">Nat Chem Biol</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/xfXfF&amp;sa=D&amp;ust=1545253715013000">&nbsp;8(10):848&ndash;854.</a></span></p><p class="c15"><span class="c14">9. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/WQRJO&amp;sa=D&amp;ust=1545253715013000">Biggs MB, Papin JA (2017) Managing uncertainty in metabolic network structure and improving predictions using EnsembleFBA. </a></span><span class="c2"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/WQRJO&amp;sa=D&amp;ust=1545253715014000">PLoS Comput Biol</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/WQRJO&amp;sa=D&amp;ust=1545253715014000">&nbsp;13(3):e1005413.</a></span></p><p class="c15"><span class="c14">10. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/m1GF6&amp;sa=D&amp;ust=1545253715014000">Dietterich TG (2000) Ensemble Methods in Machine Learning. </a></span><span class="c2"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/m1GF6&amp;sa=D&amp;ust=1545253715014000">Multiple Classifier Systems</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/m1GF6&amp;sa=D&amp;ust=1545253715014000">&nbsp;(Springer Berlin Heidelberg), pp 1&ndash;15.</a></span></p><p class="c15"><span class="c14">11. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/DKZdJ&amp;sa=D&amp;ust=1545253715015000">Kuepfer L, Peter M, Sauer U, Stelling J (2007) Ensemble modeling for analysis of cell signaling dynamics. </a></span><span class="c2"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/DKZdJ&amp;sa=D&amp;ust=1545253715015000">Nat Biotechnol</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/DKZdJ&amp;sa=D&amp;ust=1545253715015000">&nbsp;25(9):1001&ndash;1006.</a></span></p><p class="c15"><span class="c14">12. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/kA6tI&amp;sa=D&amp;ust=1545253715015000">Tran LM, Rizk ML, Liao JC (2008) Ensemble modeling of metabolic networks. </a></span><span class="c2"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/kA6tI&amp;sa=D&amp;ust=1545253715016000">Biophys J</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/kA6tI&amp;sa=D&amp;ust=1545253715016000">&nbsp;95(12):5606&ndash;5617.</a></span></p><p class="c15"><span class="c14">13. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/beLlp&amp;sa=D&amp;ust=1545253715016000">Plata G, Henry CS, Vitkup D (2015) Long-term phenotypic evolution of bacteria. </a></span><span class="c2"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/beLlp&amp;sa=D&amp;ust=1545253715017000">Nature</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/beLlp&amp;sa=D&amp;ust=1545253715017000">&nbsp;517(7534):369&ndash;372.</a></span></p><p class="c15"><span class="c14">14. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/HJkaR&amp;sa=D&amp;ust=1545253715017000">Breiman L (2001) Random Forests. </a></span><span class="c2"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/HJkaR&amp;sa=D&amp;ust=1545253715017000">Mach Learn</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/HJkaR&amp;sa=D&amp;ust=1545253715018000">&nbsp;45(1):5&ndash;32.</a></span></p><p class="c15"><span class="c14">15. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/cA6YJ&amp;sa=D&amp;ust=1545253715018000">Dong X, Quinn PJ, Wang X (2011) Metabolic engineering of Escherichia coli and Corynebacterium glutamicum for the production of L-threonine. </a></span><span class="c2"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/cA6YJ&amp;sa=D&amp;ust=1545253715018000">Biotechnol Adv</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/cA6YJ&amp;sa=D&amp;ust=1545253715019000">&nbsp;29(1):11&ndash;23.</a></span></p><p class="c15"><span class="c14">16. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/V3SZO&amp;sa=D&amp;ust=1545253715019000">Simic P, Willuhn J, Sahm H, Eggeling L (2002) Identification of glyA (encoding serine hydroxymethyltransferase) and its use together with the exporter ThrE to increase L-threonine accumulation by Corynebacterium glutamicum. </a></span><span class="c2"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/V3SZO&amp;sa=D&amp;ust=1545253715019000">Appl Environ Microbiol</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/V3SZO&amp;sa=D&amp;ust=1545253715020000">&nbsp;68(7):3321&ndash;3327.</a></span></p><p class="c15"><span class="c14">17. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/2Olqk&amp;sa=D&amp;ust=1545253715020000">Chaudhuri RR, et al. (2009) Comprehensive identification of essential Staphylococcus aureus genes using Transposon-Mediated Differential Hybridisation (TMDH). </a></span><span class="c2"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/2Olqk&amp;sa=D&amp;ust=1545253715020000">BMC Genomics</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/2Olqk&amp;sa=D&amp;ust=1545253715020000">&nbsp;10:291.</a></span></p><p class="c15"><span class="c14">18. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/C2qPA&amp;sa=D&amp;ust=1545253715021000">Akerley BJ, et al. (2002) A genome-scale analysis for identification of genes required for growth or survival of Haemophilus influenzae. </a></span><span class="c2"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/C2qPA&amp;sa=D&amp;ust=1545253715021000">Proc Natl Acad Sci U S A</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/C2qPA&amp;sa=D&amp;ust=1545253715021000">&nbsp;99(2):966&ndash;971.</a></span></p><p class="c15"><span class="c14">19. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/goaRi&amp;sa=D&amp;ust=1545253715022000">Hatzimanikatis V, et al. (2005) Exploring the diversity of complex metabolic networks. </a></span><span class="c2"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/goaRi&amp;sa=D&amp;ust=1545253715022000">Bioinformatics</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/goaRi&amp;sa=D&amp;ust=1545253715022000">&nbsp;21(8):1603&ndash;1609.</a></span></p><p class="c15"><span class="c14">20. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/LPxAy&amp;sa=D&amp;ust=1545253715023000">Jeffryes JG, et al. (2015) MINEs: open access databases of computationally predicted enzyme promiscuity products for untargeted metabolomics. </a></span><span class="c2"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/LPxAy&amp;sa=D&amp;ust=1545253715023000">J Cheminform</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/LPxAy&amp;sa=D&amp;ust=1545253715023000">&nbsp;7:44.</a></span></p><p class="c15"><span class="c14">21. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/PjDgh&amp;sa=D&amp;ust=1545253715024000">He J, et al. (2018) Quantifying Expected Uncertainty Reduction and Value of Information Using Ensemble-Variance Analysis. </a></span><span class="c2"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/PjDgh&amp;sa=D&amp;ust=1545253715024000">SPE Journal</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/PjDgh&amp;sa=D&amp;ust=1545253715024000">. Available at: </a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=https://www.onepetro.org/journal-paper/SPE-182609-PA&amp;sa=D&amp;ust=1545253715025000">https://www.onepetro.org/journal-paper/SPE-182609-PA</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/PjDgh&amp;sa=D&amp;ust=1545253715025000">.</a></span></p><p class="c15"><span class="c14">22. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/xU60c&amp;sa=D&amp;ust=1545253715025000">Monk J, Nogales J, Palsson BO (2014) Optimizing genome-scale network reconstructions. </a></span><span class="c2"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/xU60c&amp;sa=D&amp;ust=1545253715025000">Nat Biotechnol</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/xU60c&amp;sa=D&amp;ust=1545253715026000">&nbsp;32(5):447&ndash;452.</a></span></p><p class="c15"><span class="c14">23. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/5DSzt&amp;sa=D&amp;ust=1545253715026000">Magn&uacute;sd&oacute;ttir S, et al. (2017) Generation of genome-scale metabolic reconstructions for 773 members of the human gut microbiota. </a></span><span class="c2"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/5DSzt&amp;sa=D&amp;ust=1545253715026000">Nat Biotechnol</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/5DSzt&amp;sa=D&amp;ust=1545253715027000">&nbsp;35(1):81&ndash;89.</a></span></p><p class="c15"><span class="c14">24. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/rGwAW&amp;sa=D&amp;ust=1545253715027000">Wattam AR, et al. (2017) Improvements to PATRIC, the all-bacterial Bioinformatics Database and Analysis Resource Center. </a></span><span class="c2"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/rGwAW&amp;sa=D&amp;ust=1545253715027000">Nucleic Acids Res</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/rGwAW&amp;sa=D&amp;ust=1545253715027000">&nbsp;45(D1):D535&ndash;D542.</a></span></p><p class="c15"><span class="c14">25. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/ICKNq&amp;sa=D&amp;ust=1545253715028000">Ebrahim A, Lerman JA, Palsson BO, Hyduke DR (2013) COBRApy: COnstraints-Based Reconstruction and Analysis for Python. </a></span><span class="c2"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/ICKNq&amp;sa=D&amp;ust=1545253715028000">BMC Syst Biol</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/ICKNq&amp;sa=D&amp;ust=1545253715028000">&nbsp;7:74.</a></span></p><p class="c15"><span class="c14">26. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/OFgEz&amp;sa=D&amp;ust=1545253715030000">Mundy M, Mendes-Soares H, Chia N (2017) Mackinac: a bridge between ModelSEED and COBRApy to generate and analyze genome-scale metabolic models. </a></span><span class="c2"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/OFgEz&amp;sa=D&amp;ust=1545253715031000">Bioinformatics</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/OFgEz&amp;sa=D&amp;ust=1545253715031000">&nbsp;33(15):2416&ndash;2418.</a></span></p><p class="c15"><span class="c14">27. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/sngqz&amp;sa=D&amp;ust=1545253715031000">Lewis NE, et al. (2010) Omic data from evolved E. coli are consistent with computed optimal growth from genome&#8208;scale models. </a></span><span class="c2"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/sngqz&amp;sa=D&amp;ust=1545253715032000">Mol Syst Biol</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/sngqz&amp;sa=D&amp;ust=1545253715032000">&nbsp;6(1):390.</a></span></p><p class="c15"><span class="c14">28. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/37ibn&amp;sa=D&amp;ust=1545253715032000">Jensen K, Cardoso J (2016) Optlang: An algebraic modeling language for mathematical optimization. </a></span><span class="c2"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/37ibn&amp;sa=D&amp;ust=1545253715032000">Journal of Open Source</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/37ibn&amp;sa=D&amp;ust=1545253715033000">. Available at: </a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://www.forskningsdatabasen.dk/en/catalog/2350686580&amp;sa=D&amp;ust=1545253715033000">http://www.forskningsdatabasen.dk/en/catalog/2350686580</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/37ibn&amp;sa=D&amp;ust=1545253715033000">.</a></span></p><p class="c15"><span class="c14">29. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/v1H8r&amp;sa=D&amp;ust=1545253715033000">Pedregosa F, et al. (2011) Scikit-learn: Machine Learning in Python. </a></span><span class="c2"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/v1H8r&amp;sa=D&amp;ust=1545253715033000">J Mach Learn Res</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/v1H8r&amp;sa=D&amp;ust=1545253715034000">&nbsp;12(Oct):2825&ndash;2830.</a></span></p><p class="c15"><span class="c14">30. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/2g437&amp;sa=D&amp;ust=1545253715034000">Elkan C (2003) Using the triangle inequality to accelerate k-means. </a></span><span class="c2"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/2g437&amp;sa=D&amp;ust=1545253715034000">Proceedings of the 20th International Conference on Machine Learning (ICML-03)</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/2g437&amp;sa=D&amp;ust=1545253715034000">, pp 147&ndash;153.</a></span></p><p class="c15"><span class="c14">31. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/8d4f6&amp;sa=D&amp;ust=1545253715035000">Gordon AD (1984) </a></span><span class="c2"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/8d4f6&amp;sa=D&amp;ust=1545253715035000">Biometrics</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/8d4f6&amp;sa=D&amp;ust=1545253715035000">&nbsp;40(3):874&ndash;874.</a></span></p><p class="c15"><span class="c14">32. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/VWwiV&amp;sa=D&amp;ust=1545253715035000">Gower JC (1966) Some distance properties of latent root and vector methods used in multivariate analysis. </a></span><span class="c2"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/VWwiV&amp;sa=D&amp;ust=1545253715036000">Biometrika</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/VWwiV&amp;sa=D&amp;ust=1545253715036000">&nbsp;53(3-4):325&ndash;338.</a></span></p><p class="c15"><span class="c14">33. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/YxHco&amp;sa=D&amp;ust=1545253715036000">Hamming RW (1950) Error Detecting and Error Correcting Codes. </a></span><span class="c2"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/YxHco&amp;sa=D&amp;ust=1545253715036000">Bell System Technical Journal</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/YxHco&amp;sa=D&amp;ust=1545253715037000">&nbsp;29(2):147&ndash;160.</a></span></p><p class="c15"><span class="c14">34. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/XRGIp&amp;sa=D&amp;ust=1545253715037000">Chen W-H, Lu G, Chen X, Zhao X-M, Bork P (2017) OGEE v2: an update of the online gene essentiality database with special focus on differentially essential genes in human cancer cell lines. </a></span><span class="c2"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/XRGIp&amp;sa=D&amp;ust=1545253715037000">Nucleic Acids Res</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/XRGIp&amp;sa=D&amp;ust=1545253715038000">&nbsp;45(D1):D940&ndash;D944.</a></span></p><p class="c15"><span class="c14">35. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/mxooS&amp;sa=D&amp;ust=1545253715038000">Song J-H, et al. (2005) Identification of essential genes in Streptococcus pneumoniae by allelic replacement mutagenesis. </a></span><span class="c2"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/mxooS&amp;sa=D&amp;ust=1545253715038000">Mol Cells</a></span><span class="c5"><a class="c1" href="https://www.google.com/url?q=http://paperpile.com/b/7TZ51d/mxooS&amp;sa=D&amp;ust=1545253715039000">&nbsp;19(3):365&ndash;374.</a></span></p><hr style="page-break-before:always;display:none;"><p class="c9 c24"><span class="c5 c8"></span></p><p class="c24"><span class="c16 c8 c13">Supplemental Figures</span></p><p class="c24 c9"><span class="c16 c8 c13"></span></p><a id="t.944e9c7f4f944f476e54ebe51487d612599798b2"></a><a id="t.3"></a><table class="c22"><tbody><tr class="c12"><td class="c21" colspan="1" rowspan="1"><p class="c25"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 706.00px; height: 172.00px;"><img alt="" src="images/image15.png" style="width: 706.00px; height: 172.00px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p></td></tr><tr class="c12"><td class="c21" colspan="1" rowspan="1"><p class="c25"><span class="c13 c17">Supplemental Figure 1.</span><span class="c17">&nbsp;Subsampled ensemble behavior for predictions of biomass production. We simulated biomass production in a rich medium across the entire ensemble and subsampled these results at varying ensemble sizes. </span><span class="c13 c17">a)</span><span class="c17">&nbsp;Standard deviation of the mean flux through biomass from each subsample and </span><span class="c13 c17">b)</span><span class="c6 c8 c17 c27">&nbsp;standard deviation of the standard deviation of flux through biomass in each subsample. For both quantities (variance of the mean of each subsample and variance of the variance of each subsample), simulations plateau before inclusion of all 1000 ensemble members. Values on y axis are normalized by dividing by the mean flux through biomass for the entire ensemble.</span></p></td></tr></tbody></table><p class="c24 c9"><span class="c16 c8 c13"></span></p></body></html>